.github/workflows/test-and-deploy-amplify.yml

```diff
@@ -0,0 +1,59 @@
+name: Test the and deploy MVP (AWS Amplify)
+
+on: workflow_dispatch
+
+env:
+  pulumi_stack_organization: virtualfinland
+  aws_region: eu-north-1
+  PULUMI_ACCESS_TOKEN: ${{ secrets.PULUMI_ACCESS_TOKEN }}
+
+jobs:
+  test:
+    name: Test the app
+    runs-on: ubuntu-latest
+    environment: dev
+    strategy:
+      matrix:
+        node-version: [18.x]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: App install, test
+        run: |
+          npm ci
+          npm run test:ci
+  deploy:
+    name: Deployment
+    runs-on: ubuntu-latest
+    needs: test
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - name: Configure AWS credentials
+        uses: Virtual-Finland-Development/infrastructure/.github/actions/configure-aws-credentials@main
+        with:
+          environment: dev
+          aws-region: ${{ secrets.AWS_REGION }}
+          pulumi-access-token: ${{ secrets.PULUMI_ACCESS_TOKEN }}
+      - name: Install Pulumi CLI
+        uses: pulumi/actions@v4
+      - name: Install dependencies
+        working-directory: ./infra/amplify
+        run: npm ci
+      - name: Select Pulumi stack
+        working-directory: ./infra/amplify
+        run: pulumi stack select ${{ env.pulumi_stack_organization }}/amplify --create
+      - name: Set github access token in Pulumi config
+        working-directory: ./infra/amplify
+        run: pulumi config set --secret githubAccessToken ${{ secrets.PAT_TOKEN }}
```

Comment:
Tässä varmaan saattaisi toimia se ci-pipelinen automaaginen github-token: https://docs.github.com/en/actions/security-guides/automatic-token-authentication

Jos se ei aikaisemmin yrityksissä toiminut, niin ehkäpä syy siinä kun oli konffissa se .git-postfixi reponimessä.

Reply:
Se .git-postfixi ei estänyt deplausta, se vaan johti yhteysongelmiin amplifyn/githubin välillä. GITHUB_TOKEN ei kelvannut ja deplaus failasi, mutta kokeilen vielä varmuuden vuoksi uudelleen. 🕵️

Reply:
I stand corrected - meni läpi onnistuneesti nyt tuolla github-tokenilla, en tosin ymmärrä täysin että miksi se ei aikaisemmin toiminut. 🤔

Reply:
> miksi se ei aikaisemmin toiminut.
Se .git-postifxi aiheutti sen että amplify/github välinen yhteys osoitti repositoriin jota ei ole olemassa, eli sen jälkeen kun amplify oli pystytetty (deplaus onnistui) niin sen jälkeinen amplifin build-phase kaatui 401 access denied-virheeseen. Tämä virheen harhaanjohtava koodi tulee siitä että github API palauttelee 401 - access denied myös silloin kun repoa ei ole olemassa, eli sieltä ei tule 404 Not found. Eli lyhyesti amplifin deplaus onnistui mutta se branch-linkitys ei.

Reply:
`* creating Amplify App (virtual-finland-amplifyApp-amplify-4598583): BadRequestException: There was an issue setting up your repository. Please try again later.({"message":"Resource not accessible by integration","documentation_url":"https://docs.github.com/rest/webhooks/repos#create-a-repository-webhook"})`

(GITHUB_TOKEN)

`creating Amplify App (virtual-finland-amplifyApp-amplify-5b598d6): BadRequestException: There was an issue setting up your repository. Please try again later.({"message":"Resource not accessible by personal access token","documentation_url":"https://docs.github.com/rest/webhooks/repos#create-a-repository-webhook"})`

(PAT ilman riittäviä oikeuksia)

Näitä erroreita se heitti aikaisemmin. Kuitenkin sitten kun lokaalista deplaili pulumilla suoraan (PAT-tokenilla), deplaus meni läpi ilman virheitä (edelleen .git-postfixin kanssa), mutta itse buildaus failasi amplifyn puolella.

Tuo selitys käy kyllä järkeen. 👍  Hämyä tuo että se kuitenkin deplaili mukisematta lokaalisti uppia laittamalla, mutta ei actionin kautta. Oli miten oli, todettu nyt että toimii. 👍